3d_armor_stand/init.lua

```diff
@@ -149,236 +149,137 @@ minetest.register_node("3d_armor_stand:top", {
+	local function owns_armor_stand(pos, meta, player)
+		if (def.name == "locked_armor_stand" and not has_locked_armor_stand_privilege(meta, player)) or
+		   (def.name == "shared_armor_stand" and (not minetest.is_player(player) or
+		   minetest.is_protected(pos, player:get_player_name()))) then
+			return false
 		end
 		return true
-	end,
-	after_place_node = function(pos, placer)
-		minetest.add_entity(pos, "3d_armor_stand:armor_entity")
-		add_hidden_node(pos, placer)
-	end,
-	allow_metadata_inventory_put = function(pos, listname, index, stack)
-		local def = stack:get_definition() or {}
-		local groups = def.groups or {}
-		if groups[listname] then
-			return 1
-		end
-		return 0
-	end,
-	allow_metadata_inventory_move = function(pos)
-		return 0
-	end,
-	on_metadata_inventory_put = function(pos)
-		update_entity(pos)
-	end,
-	on_metadata_inventory_take = function(pos)
-		update_entity(pos)
-	end,
-	after_destruct = function(pos)
-		update_entity(pos)
-		remove_hidden_node(pos)
-	end,
-	on_blast = function(pos)
-		drop_armor(pos)
-		armor.drop_armor(pos, "3d_armor_stand:armor_stand")
-		minetest.remove_node(pos)
-	end,
-})
+	end
 
-minetest.register_node("3d_armor_stand:locked_armor_stand", {
-	description = S("Locked Armor Stand"),
-	drawtype = "mesh",
-	mesh = "3d_armor_stand.obj",
-	tiles = {"3d_armor_stand_locked.png"},
-	use_texture_alpha = "clip",
-	paramtype = "light",
-	paramtype2 = "facedir",
-	walkable = false,
-	selection_box = {
-		type = "fixed",
-		fixed = {
-			{-0.25, -0.4375, -0.25, 0.25, 1.4, 0.25},
-			{-0.5, -0.5, -0.5, 0.5, -0.4375, 0.5},
+	minetest.register_node("3d_armor_stand:" .. def.name, {
+		description = S(def.description),
+		drawtype = "mesh",
+		mesh = "3d_armor_stand.obj",
+		tiles = {def.texture},
+		use_texture_alpha = "clip",
+		paramtype = "light",
+		paramtype2 = "facedir",
+		walkable = false,
+		selection_box = {
+			type = "fixed",
+			fixed = {
+				{-0.25, -0.4375, -0.25, 0.25, 1.4, 0.25},
+				{-0.5, -0.5, -0.5, 0.5, -0.4375, 0.5},
+			},
 		},
-	},
-	groups = {choppy=2, oddly_breakable_by_hand=2},
-	is_ground_content = false,
-	sounds = default.node_sound_wood_defaults(),
-	on_construct = function(pos)
-		local meta = minetest.get_meta(pos)
-		meta:set_string("formspec", armor_stand_formspec)
-		meta:set_string("infotext", S("Armor Stand"))
-		meta:set_string("owner", "")
-		local inv = meta:get_inventory()
-		for _, element in pairs(elements) do
-			inv:set_size("armor_"..element, 1)
-		end
-	end,
-	can_dig = function(pos, player)
-		local meta = minetest.get_meta(pos)
-		local inv = meta:get_inventory()
-		for _, element in pairs(elements) do
-			if not inv:is_empty("armor_"..element) then
+		groups = {choppy=2, oddly_breakable_by_hand=2},
+		is_ground_content = false,
+		sounds = default.node_sound_wood_defaults(),
+		on_construct = function(pos)
+			local meta = minetest.get_meta(pos)
+			meta:set_string("formspec", armor_stand_formspec)
+			meta:set_string("infotext", S(def.description))
```

Comment:
This is not picked up by the translator script. Furthermore, `def.description` should be translated. `S` should not belong here.

Reply:
It's the one mentioned in lua_api.md
https://github.com/luanti-org/modtools/blob/main/mod_translation_updater.py